lib/iris/__init__.py

```diff
@@ -303,7 +315,301 @@ def _load_collection(uris, constraints=None, callback=None):
+        so will take precedence.
+
+        """
+        if options is None:
+            options = {}
+        elif isinstance(options, str) and options in self.SETTINGS:
+            options = self.SETTINGS[options]
+        elif not isinstance(options, Mapping):
+            msg = (
+                f"Invalid arg options='{options!r}' : "
+                f"must be a dict, or one of {self.SETTINGS.keys()}"
+            )
+            raise ValueError(msg)
+
+        # Override any options with keywords
+        options.update(**kwargs)
+        bad_keys = [key for key in options if key not in self.OPTION_KEYS]
+        if bad_keys:
+            msg = f"Unknown options {bad_keys} : valid options are {self.OPTION_KEYS}."
+            raise ValueError(msg)
+
+        # Implement all options by changing own content.
+        for key, value in options.items():
+            setattr(self, key, value)
+
+    def settings(self):
+        """Return an options dict containing the current settings."""
+        return {key: getattr(self, key) for key in self.OPTION_KEYS}
+
+    def __repr__(self):
+        msg = f"{self.__class__.__name__}("
+        msg += ", ".join(f"{key}={getattr(self, key)!r}" for key in self.OPTION_KEYS)
+        msg += ")"
+        return msg
+
+    @contextlib.contextmanager
+    def context(self, settings=None, **kwargs):
+        """Return a context manager applying given options.
+
+        Parameters
+        ----------
+        settings : str or dict
+            Options dictionary or name, as for :meth:`~LoadPolicy.set`.
+        kwargs : dict
+            Option values, as for :meth:`~LoadPolicy.set`.
+
+        Examples
+        --------
+        .. testsetup::
+
+            import iris
+            from iris import LOAD_POLICY, sample_data_path
+            path = sample_data_path("hybrid_height.nc")
+
+        >>> with LOAD_POLICY.context("comprehensive"):
+        ...     cubes = iris.load(path)
+
+        """
+        # Save the current state
+        saved_settings = self.settings()
+
+        # Apply the new options and execute the context
+        try:
+            self.set(settings, **kwargs)
+            yield
+        finally:
+            # Re-establish the former state
+            self.set(saved_settings)
+
+
+#: Object containing file loading options.
+LOAD_POLICY = LoadPolicy("legacy")
+# The unique (singleton) policy object
+# N.B. FOR NOW, our starting point is "legacy" rather than "default"
+# TODO: resolve tests as needed, to pass with "default".
+
+
+def combine_cubes(cubes, options=None, merge_require_unique=False):
+    """Combine cubes as for load, according to "loading policy" options.
+
+    Applies :meth:`~iris.cube.CubeList.merge`/:meth:`~iris.cube.CubeList.concatenate`
+    steps to the given cubes, as determined by the 'settings'.
+
+    Parameters
+    ----------
+    cubes : list of :class:`~iris.cube.Cube`
+        A list of cubes to combine.
```

Comment:
It looks like, as written, `cubes` would be required to be a CubeList. I think it would actually be quite neat if this was allowed to take in a list of cubes as well as a CubeList, though I think there should be some appropriate error checking logic added for this.

Reply:
Yes, I have just fixed this while writing tests (subsequent PR).
In future, it will convert an input ordinary list to a CubeList, and always returns a CubeList.